From 1da4a16ae9844f0651a9f6d28fd4bf679bac6d9a Mon Sep 17 00:00:00 2001
From: Francesco Pham <francesco.pham1@h-partners.com>
Date: Thu, 28 May 2026 11:59:42 +0200
Subject: [PATCH] profiler: x86_64 native_hook regs

native_hook source set adds the x86_64 getcontext assembly. (Upstream
OpenHarmony-7.0 already picks DfxRegsX86_64 in native_daemon and declares
the x86_64 FULL_PATH socket constants.)

Change-Id: I058c2953056b4af937adb6ce85b940d0cf4cc6e4
Signed-off-by: Francesco Pham <francesco.pham1@h-partners.com>
---
 device/plugins/native_hook/BUILD.gn | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/device/plugins/native_hook/BUILD.gn b/device/plugins/native_hook/BUILD.gn
index 7a5cab0d8..8f3a7f6cd 100644
--- a/device/plugins/native_hook/BUILD.gn
+++ b/device/plugins/native_hook/BUILD.gn
@@ -57,6 +57,11 @@ ohos_source_set("native_hook_source") {
     "src/stack_writer.cpp",
     "src/cache_file_writer.cpp",
   ]
+
+  if (target_cpu == "x86_64") {
+    sources += ["//base/hiviewdfx/faultloggerd/interfaces/innerkits/unwinder/src/registers/getcontext_x86_64.S"]
+  }
+
   include_dirs = [
     "./include",
     "../api/include",
-- 
2.50.1

